CONTRIBUTING.md

```diff
@@ -47,6 +47,61 @@ and additionally the following rules:
 * Use `my ($self) = @_;` for parameter parsing in methods when accessing the
   `$self` object. Do not parse any parameter if you do not need any.
 * [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself)
+* Avoid `sleep()`: Do not use `sleep()` or simulate a sleep-like functionality.
+  This is introducing the risk of either unnecessarily wasting time or just
+  shifting race conditions and making them even harder to investigate. You *can*
+  use sleep but there are only very limited cases where is the best solution
+  (e.g. yield threads in low-level system code). Here you have the classical
+  problem in software design of
+  [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science))
+  . sleep should not be used in place of proper synchronization methods.
+  https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/
+  is one of multiple references that explains this: "The original problem is
+  likely a timing/synchronization issue, ignoring it by hiding it with
+  Thread.Sleep is only going to delay the problem and make it occur in random,
+  hard to reproduce ways." And if the problem does not occur at least time is
+  wasted.
+  We are waiting because some condition changes some time ... keyword(s) is/are
+  *some time*. It is important to know the exact condition that the code should
+  wait for and check for that instead of delaying program execution for an
+  arbitrary amount of time. For openQA tests in general just try to do what you
+  as a human user would also do. You would not look on your watch waiting for
+  exactly 100s before you look at the screen again, right? :)
+* Avoid `is_tumbleweed()`: Our tests should always consider openSUSE Tumbleweed
+  as the default product without a version. That means any differing behaviour
+  should have an explicit exclude rule for the "older products" and potentially
+  an explanation why the behaviour should differ. You can try to negate the logic
+  check.
+* Consider a corresponding Tumbleweed test: Because of
```

Comment:
I'd say that just `considering` is too weak. I've seen numerous times where tests were added which only targeted SLE for no reason whatsoever. What about "The test must also run on Tumbleweed unless ..."?

Reply:
Well, with "consider" I meant to thoroughly think about if this applies or not. The following line already states a pretty strong "we should test against Tumbleweed by default (with exceptions…)". And a plead after that even with "So please check …". I consider – here used in different meaning ;) – your wording to be inconsistent with the style of using an imperative word in most other practices. Thinking about it I am not sure what is better, I thought of "Conduct a corresponding Tumbleweed test" but I like it less because there can be good reasons where a Tumbleweed test is not applicable as explained in the text and I don't want to water down the rules by stating a rule that can not be followed when it says "Conduct …".

Reply:
> Well, with "consider" I meant to thoroughly think about if this applies or not.

I get that, but the "thouroughly" part is not really written down here. Currently it seems more like a "If you have time, please also ..." to me.

> The following line already states a pretty strong "we should test against Tumbleweed by default (with exceptions…)"

A "pretty strong" "should"? :P

> I consider – here used in different meaning ;) – your wording to be inconsistent with the style of using an imperative word in most other practices.

Your diff starts with "Avoid `sleep()`". That's IMO much stronger than "consider" or "should" already.

> Thinking about it I am not sure what is better, I thought of "Conduct a corresponding Tumbleweed test" but I like it less because there can be good reasons where a Tumbleweed test is not applicable as explained in the text and I don't want to water down the rules by stating a rule that can not be followed when it says "Conduct …".

I agree, but a "unless you have a good reason, like software not being in Tumbeweed or the case not being relevant otherwise" ought to take care of that. If it's not possible to put this in words, then it'll be hard to get everyone to agree and follow it.

Reply:
Yes, you are right. Trust me, I will create a specific PR to clarify that as I don't want to "sneak in" policy changes when the PR has already been approved by so many in the current state.

Reply:
please see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11546/commits/6d30a33eea2ec88f450f7772869beaebbedb1256